crates/driver/src/boundary/notifier/liquorice/mod.rs

```diff
@@ -0,0 +1,135 @@
+    settlement_contract_address: eth::Address,
+}
+
+impl LiquoriceNotifier {
+    pub fn new(
+        config: &infra::notify::liquidity_sources::config::Liquorice,
+        chain_id: u64,
+    ) -> Result<Self> {
+        let settlement_contract_address = ILiquoriceSettlement::raw_contract()
+            .networks
+            .get(chain_id.to_string().as_str())
+            .map(|network| network.address.into())
+            .ok_or(anyhow!("Liquorice settlement contract not found"))?;
+
+        Ok(Self {
+            liquorice_api: DefaultLiquoriceApi::new(
+                reqwest::ClientBuilder::default(),
+                config.base_url.clone(),
+                config.api_key.clone(),
+                config.http_timeout,
+            )?,
+            settlement_contract_address,
+        })
+    }
+
+    /// Extracts Liquorice maker RFQ IDs from the settlement interactions.
+    /// <https://liquorice.gitbook.io/liquorice-docs/for-market-makers/basic-market-making-api#id-3.-receiving-rfq>
+    fn extract_rfq_ids_from_settlement(&self, settlement: &Settlement) -> Result<HashSet<String>> {
+        // Aggregate all interactions from the settlement
+        let pre_interactions = settlement.pre_interactions();
+        let interactions = settlement
+            .interactions()
+            .iter()
+            .filter_map(|interaction| match interaction {
+                solution::Interaction::Custom(custom) => Some(eth::Interaction {
+                    target: custom.target.into(),
+                    value: custom.value,
+                    call_data: custom.call_data.clone(),
+                }),
+                solution::Interaction::Liquidity(_) => None,
+            })
+            .collect::<Vec<eth::Interaction>>();
+        let post_interactions = settlement.post_interactions();
+
+        // Extract RFQ IDs from the interactions
+        let rfq_ids = pre_interactions
+            .iter()
+            .chain(interactions.iter())
+            .chain(post_interactions.iter())
+            .filter(|interaction| interaction.target == self.settlement_contract_address)
+            .map(|interaction| Self::extract_rfq_id_from_calldata(&interaction.call_data))
+            .collect::<Result<Vec<String>>>()?;
+
+        Ok(HashSet::from_iter(rfq_ids.into_iter()))
+    }
+
+    fn extract_rfq_id_from_calldata(calldata: &Bytes<Vec<u8>>) -> Result<String> {
+        // Decode the calldata using the Liquorice settlement contract ABI
+        let settle_single_function = ILiquoriceSettlement::raw_contract()
+            .interface
+            .abi
+            .function("settleSingle")
+            .unwrap();
+        let tokens = settle_single_function
+            .decode_output(calldata.0.as_slice())
+            .context("Invalid output from settleSingle")?;
+
+        // Token at index 1 is expected to be a tuple corresponding to Liquorice
+        // Settlement single Order
```

Comment:
Could you link to the solidity interface here. That's just easier to review then.

Reply:
Done, provided [links](https://github.com/cowprotocol/services/blob/3014e1d6d387108c39b6c3ead30a2999ca3f737b/crates/driver/src/infra/notify/liquidity_sources/liquorice/notifier.rs#L153) to the contract's code at Etherscan